src/ui/overlayService/overlayService.js

```diff
@@ -43,45 +75,137 @@ define([
+    };
+
+    /**
+     * A description of the model options that may be passed to the
+     * showBlockingMessage method. Note that the DialogModel described
+     * here is shared with the Notifications framework.
+     * @see NotificationService
+     *
+    * @typedef model
+        * @property {string} title the title to use for the dialog
+        * @property {string} severity the severity level of this message.
+        * These are defined in a bundle constant with key 'dialogSeverity'
+        * @property {string} hint the 'hint' message to show below the title
+        * @property {string} actionText text that indicates a current action,
+        * shown above a progress bar to indicate what's happening.
+        * @property {number} progress a percentage value (1-100)
+        * indicating the completion of the blocking task
+        * @property {string} progressText the message to show below a
+        * progress bar to indicate progress. For example, this might be
+        * used to indicate time remaining, or items still to process.
+        * @property {boolean} unknownProgress some tasks may be
+        * impossible to provide an estimate for. Providing a true value for
+        * this attribute will indicate to the user that the progress and
+        * duration cannot be estimated.
+        * @property {Object} primaryOption an action that will
+        * be added to the dialog as a button. The primary action can be
+        * used as the suggested course of action for the user. Making it
+        * distinct from other actions allows it to be styled differently,
+        * and treated preferentially in banner mode.
+        * @property {buttons[]} buttons a list of buttons with title and callback properties that will
+        * be added to the dialog.
+    */
 
-        lastActiveComponent.$destroy(true);
-        document.body.removeChild(lastActiveOverlay);
+    /**
+     * Displays a blocking (modal) dialog. This dialog can be used for
+     * displaying messages that require the user's
+     * immediate attention. The message may include an indication of
+     * progress, as well as a series of actions that
+     * the user can take if necessary
+     * @param {model} dialogModel defines options for the dialog
+     * @returns {object} with a dismiss function that can be called from the calling code
+     * to dismiss/destroy the dialog
+    */
+    OverlayService.prototype.showBlockingMessage = function (model) {
+        let component = new Vue({
+            provide: {
+                model: model
+            },
+            components: {
+                BlockingMessage: BlockingMessage.default
+            },
+            template: '<blocking-message></blocking-message>'
+        });
 
-        if (this.activeOverlays.length) {
-            this.activeOverlays[this.activeOverlays.length - 1].overlay.classList.remove('invisible');
+        function destroy() {
+            component.$destroy(true);
```

Comment:
Passing a boolean to $destroy [was removed in Vue 2.x](https://github.com/vuejs/vue/issues/3534#issuecomment-242813756), we shouldn't use it.